pkg/network/opencontrail/network.go

```diff
@@ -123,6 +118,65 @@ func (m *NetworkManagerImpl) DeleteFloatingIpPool(network *types.VirtualNetwork,
 	return nil
 }
 
+// Subnets can only be deleted if there are no instance-ips or floating-ips associated
+// with the subnet.
+func (m *NetworkManagerImpl) updateSubnetConfig(network *types.VirtualNetwork, prefix string) {
```

Comment:
It does check it. But i believe it is preferable if the network-manager knows what is going on and what are the right semantics and doesn't issue invalid requests. To really clean up the prefix it will have to re-issue the command at a latter stage when the condition for delete are no longer false.